### DISCUSSION TITLE
CGNAT and PiVPN

### DISCUSSION THREAD

**Discussion by Geffers** [pivpn/pivpn#1865] [Support]
Folks, I've run PiVPN for a few years now with no issues, recently changed ISP (Community Fibre)  and apparently they run their system behind CGNAT.  Was unaware of CGNAT before this issue and am now wondering what the best route may be to continue using PiVPN as it appears with CGNAT port redirection does not work so unable to make my home network visible to the internet.

My main reason for using PiVPN was to make my home network available to me so installing PiVPN on a cloud system loses that function.

Geoff

**Comment by orazioedoardo** [comment]
If you are unable to get a public IP address, you could install PiVPN on a cloud server and route your traffic to the home network through it but it's not as easy as PiVPN. Alternatively, you could use something like Tailscale.

**Reply by Geffers** [reply]
Tried ssh -R 1194:localhost:1194 user@cloudserver but keep getting 

> Warning: remote port forwarding failed for listen port 1194

I've tried changing to port 2194 get same message but relates to 2194 failed to listen.  I've changed GatewayPorts to yes in /etc/ssh/sshd_config and still get same **Failed for listen port**

I do have access to a cloud Ubuntu system with Oracle, hence my attempts at reverse proxy but so far not working.

I'll look in to Tailscale and ngrok but I'd prefer to not use 3rd party apps if possible, at least my Oracle system is my setup.

Geoff

**Reply by Geffers** [reply]
Have overcome the failure of listening port.  Also need to use a TCP connection but so far it works.

Geffers
